contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol

```diff
@@ -0,0 +1,499 @@
+            _state().owner = _newOwner;
+            emit OwnershipTransferred(_owner, _newOwner);
+        }
+    }
+
+
+    // ================================================================================================================
+    // --- Full implementation of 'IWitnetRequestBoardAdminACLs' ------------------------------------------------------
+
+    /// Tells whether given address is included in the active reporters control list.
+    /// @param _reporter The address to be checked.
+    function isReporter(address _reporter) public view override returns (bool) {
+        return _acls().isReporter_[_reporter];
+    }
+
+    /// Adds given addresses to the active reporters control list.
+    /// @dev Can only be called from the owner address.
+    /// @dev Emits the `ReportersSet` event. 
+    /// @param _reporters List of addresses to be added to the active reporters control list.
+    function setReporters(address[] memory _reporters)
+        public
+        override
+        onlyOwner
+    {
+        for (uint ix = 0; ix < _reporters.length; ix ++) {
+            address _reporter = _reporters[ix];
+            _acls().isReporter_[_reporter] = true;
+        }
+        emit ReportersSet(_reporters);
+    }
+
+    /// Removes given addresses from the active reporters control list.
+    /// @dev Can only be called from the owner address.
+    /// @dev Emits the `ReportersUnset` event. 
+    /// @param _exReporters List of addresses to be added to the active reporters control list.
+    function unsetReporters(address[] memory _exReporters)
+        public
+        override
+        onlyOwner
+    {
+        for (uint ix = 0; ix < _exReporters.length; ix ++) {
+            address _reporter = _exReporters[ix];
+            _acls().isReporter_[_reporter] = false;
+        }
+        emit ReportersUnset(_exReporters);
+    }
+
+
+    // ================================================================================================================
+    // --- Full implementation of 'IWitnetRequestBoardReporter' -------------------------------------------------------
+
+    /// Reports the Witnet-provided result to a previously posted request. 
+    /// @dev Will asume `block.number` as the epoch number for the provided result.
+    /// @dev Fails if:
+    /// @dev - the `_queryId` is not in 'Posted' status.
+    /// @dev - provided `_proof` is zero;
+    /// @dev - length of provided `_result` is zero.
+    /// @param _queryId The unique identifier of the data request.
+    /// @param _proof of the solving tally transaction in Witnet.
+    /// @param _result The result itself as bytes.
+    function reportResult(
+            uint256 _queryId,
+            bytes32 _proof,
+            bytes calldata _result
+        )
+        external
+        override
+        onlyReporters
+        inStatus(_queryId, Witnet.QueryStatus.Posted)
+    {
+        _reportResult(_queryId, block.number, _proof, _result);
```

Comment:
Since the epoch is needed for the data feed (but not the block number), we could set the epoch to 0 in this case, so that the data feeds knows that when is 0, but has proof != 0, has to look for the epoch using the timestamp of the transaction.  Otherwise it should detect somehow that the epoch provided is not a real epoch but a block number

Reply:
The problem with that approach is that the "data feed [contract]" would have to guess the "timestamp" from the transaction that ultimately called to either `readResult` or `deleteQuery`, not from the one that called `reportResult`, which in all cases would be closer to the real and factual epoch in which the Data Request was actually solved by Witnet. 

`epoch` refers not to a timestamp, but to a block "height" in all cases.

Reply:
Better still, instead of `epoch` let reporters directly specify the Witnet-provided timestamp, in which case `Response.epoch` would be no longer needed.